pkg/pipeline/CiCdPipelineOrchestrator.go

```diff
@@ -1579,3 +1641,138 @@ func (impl CiCdPipelineOrchestratorImpl) CreateEcrRepo(dockerRepository, AWSRegi
 	}
 	return nil
 }
+
+func (impl CiCdPipelineOrchestratorImpl) InitiateMigrationOfStageScriptsToPipelineStageSteps(cdPipeline *bean.CDPipelineConfigObject) (*bean.CDPipelineConfigObject, error) {
+	//below implementation means that preStageConfig or postStageConfig is not yet migrated to pipeline_stage_steps
+	//so convert the preStageConfig or postStageConfig into pipeline_stage_steps via adapter
+	if cdPipeline.PreDeployStage == nil && len(cdPipeline.PreStage.Config) > 0 {
+		preDeployStageConverted, err := StageYamlToPipelineStageAdapter(cdPipeline.PreStage.Config, repository5.PIPELINE_STAGE_TYPE_PRE_CD, cdPipeline.PreStage.TriggerType)
+		if err != nil {
+			impl.logger.Errorw("error in converting pre-stage data into pipeline stage steps", "err", err, "cdPipelineId", cdPipeline.Id)
+			return nil, err
+		}
+		cdPipeline.PreDeployStage = preDeployStageConverted
+	}
+	if cdPipeline.PostDeployStage == nil && len(cdPipeline.PostStage.Config) > 0 {
+		postDeployStageConverted, err := StageYamlToPipelineStageAdapter(cdPipeline.PostStage.Config, repository5.PIPELINE_STAGE_TYPE_POST_CD, cdPipeline.PostStage.TriggerType)
+		if err != nil {
+			impl.logger.Errorw("error in converting post-stage data into pipeline stage steps", "err", err, "cdPipelineId", cdPipeline.Id)
+			return nil, err
+		}
+		cdPipeline.PostDeployStage = postDeployStageConverted
+	}
+	return cdPipeline, nil
+
+}
+
+func (impl CiCdPipelineOrchestratorImpl) StageStepsToCdStageAdapter(deployStage *bean2.PipelineStageDto) (*bean.CdStage, error) {
+	cdStage := &bean.CdStage{
+		Name:        deployStage.Name,
+		TriggerType: deployStage.TriggerType,
+	}
+	cdPipelineConfig := make([]CdPipelineConfig, 0)
+	beforeTasks := make([]*Task, 0)
+	afterTasks := make([]*Task, 0)
+	for _, step := range deployStage.Steps {
+		if step.InlineStepDetail != nil {
+			if deployStage.Type == repository5.PIPELINE_STAGE_TYPE_PRE_CD {
+				beforeTask := &Task{
+					Name:           step.Name,
+					Script:         step.InlineStepDetail.Script,
+					OutputLocation: strings.Join(step.OutputDirectoryPath, ","),
+				}
+				beforeTasks = append(beforeTasks, beforeTask)
+
+			}
+			if deployStage.Type == repository5.PIPELINE_STAGE_TYPE_POST_CD {
+				afterTask := &Task{
+					Name:           step.Name,
+					Script:         step.InlineStepDetail.Script,
+					OutputLocation: strings.Join(step.OutputDirectoryPath, ","),
+				}
+				afterTasks = append(afterTasks, afterTask)
+			}
+
+		} else {
+			return nil, nil
```

Comment:
please discuss if we need to throw an error in this case or not

Reply:
okay, will discuss

Reply:
TBD

Reply:
throw error here :- data is incompatible with v1 (discuss with UX)